### PR TITLE
wolfMQTT Release v1.14.1 preparation

### PR DESCRIPTION
**Pull request by embhorn**
Also revert vs_settings.h change in #305

### CODE REVIEW

File: wolfmqtt/vs_settings.h [hunk 1]
```diff
@@ -4,9 +4,8 @@
 #ifndef _WOLFMQTT_VS_SETTINGS_
 #define _WOLFMQTT_VS_SETTINGS_
 
-/* Don't include this if using autoconf cross-compile or cmake */
-#if !defined(HAVE_CONFIG_H) && !defined(BUILDING_CMAKE)
-
+/* Don't include this if using autoconf cross-compile */
+#if !defined(HAVE_CONFIG_H)
```
Comment (dgarske) [inline wolfmqtt/vs_settings.h:8]: This `BUILDING_CMAKE` fixed a vcpkg issue. Do you have a plan for how to fix this in vcpkg with Windows Cmake?  I was not able to get the vcpkg update through without it. See https://github.com/microsoft/vcpkg/pull/25936

**Review by dgarske — CHANGES_REQUESTED** [review]

File: wolfmqtt/vs_settings.h (hunk 1 above)
Comment (embhorn) [reply wolfmqtt/vs_settings.h:8]: Yeah, I'll have to figure something else out. That change definitely breaks vcpkg / VS builds.

Comment (embhorn) [reply wolfmqtt/vs_settings.h:8]: From zd14640
>The problem was as I explained earlier; the config used to build the wolfMQTT library in vcpkg was different from the config the application was using. I was able to resolve the issue by removing the section of the diff patch that changes wolfmqtt/vs_settings.h

Comment (dgarske) [reply wolfmqtt/vs_settings.h:8]: That's fine, but you will likely need to find another solution for the vcpkg CI tests with wolfMQTT and Windows.

**Review by dgarske — APPROVED** [review]

**Comment by chenrui333** [comment]
Just heads up, [this merge build](https://github.com/wolfSSL/wolfMQTT/runs/8123500764) needs a rerun.

**Pull request merged**